pkg/virtctl/create/vm/vm.go

```diff
@@ -646,14 +702,18 @@ func (c *createVM) getInferFromVolume(vm *v1.VirtualMachine) (string, error) {
 	return vm.Spec.Template.Spec.Volumes[0].Name, nil
 }
 
+func (c *createVM) hasCloudInitRelevantFlags() bool {
+	return c.cmd.Flags().Changed(UserFlag) ||
+		c.cmd.Flags().Changed(PasswordFileFlag) ||
+		c.cmd.Flags().Changed(SSHKeyFlag) ||
+		c.cmd.Flags().Changed(GAManageSSHFlag) ||
+		c.cmd.Flags().Changed(CloudInitFlag) ||
+		c.cmd.Flags().Changed(CloudInitUserDataFlag) ||
+		c.cmd.Flags().Changed(CloudInitNetworkDataFlag)
+}
+
 func (c *createVM) cloudInitConfig(vm *v1.VirtualMachine) error {
-	if !c.cmd.Flags().Changed(UserFlag) &&
-		!c.cmd.Flags().Changed(PasswordFileFlag) &&
-		!c.cmd.Flags().Changed(SSHKeyFlag) &&
-		!c.cmd.Flags().Changed(GAManageSSHFlag) &&
-		!c.cmd.Flags().Changed(CloudInitFlag) &&
-		!c.cmd.Flags().Changed(CloudInitUserDataFlag) &&
-		!c.cmd.Flags().Changed(CloudInitNetworkDataFlag) {
+	if !c.hasCloudInitRelevantFlags() {
```

Comment:
Was it a linter suggestion?

Reply:
yes, cyclomatic complexity 21 of func (*createVM).cloudInitConfig is high (> 15) (gocyclo).
So, I've moved relevant flags to outer function.